OCast/ReferenceDriver/Driver/ReferenceDriver.swift

```diff
@@ -177,7 +176,14 @@ import Foundation
             links[module] = link
             connectCallbacks[module] = Callback(success: onSuccess, error: onError)
             linksState[module] = .connecting
-            link?.connect()
+            guard link?.connect() ?? false else {
+                links.removeValue(forKey: module)
+                connectCallbacks.removeValue(forKey: module)
+                linksState[module] = .disconnected
+                let error = NSError(domain: ReferenceDriver.referenceDriverErrorDomain, code: ReferenceDriverErrorCode.moduleNotConnected.rawValue, userInfo: nil)
```

Comment:
new error code : bad application url

Reply:
only application url can throw this error ?